### PR TITLE
Extended source

### PR DESCRIPTION
**Pull request by Crauzerus**
PR just for the review - I will not merge it

### CODE REVIEW

**Review by wkrzemien — COMMENTED** [review]
Is the file t14_extended_source/output.png added as reference?

File: t14_extended_source/runAnalysis.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  Method calls by runBenchmark script.
+
+  Parameters
+  ----------
+  output_folders : list
+   list of output folders
+
+  Returns
+  -------
+  benchark finihed without errors : bool
```
Comment (wkrzemien) [inline t14_extended_source/runAnalysis.py]:
```suggestion
  benchmark finished without errors : bool
```

File: t14_extended_source/runAnalysis.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    csvr = csv.reader(file)
+    next(csvr)
+    test_edeps = list()
+    for edep in csvr:
+      test_edeps.append(float(edep[0]))
+    test_edeps = np.array(test_edeps, dtype=np.float32)
+  _, pvalue = stats.ks_2samp(edeps, test_edeps)
+  #in scipy.stats.ks_2samp a hypothesis is for two-sided test is: two distributions are identical, F(x)=G(x) for all x
+  #We can say with certainty that our hypothesis is not true for pvalue equals 5% or lower.
+  pvalue_threshold = 0.05
```
Comment (wkrzemien) [inline t14_extended_source/runAnalysis.py]: I would propose to pass pvalue_theshold as an argument to this function

File: t14_extended_source/runAnalysis.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+  gamma_type_strings = {GammaType.ANNIHILATIONGAMMA:"-annihilation-gamma",GammaType.PROMPTGAMMA:"-prompt-gamma"}
+  script_dir_path = Path(os.path.dirname(os.path.abspath(__file__)))
+  tests_data_dir_path = script_dir_path/"data"
+  file_path = tests_data_dir_path/"".join([source_strings[source_type],decay_strings[decay_type],gamma_type_strings[gamma_type],".csv"])
+  with file_path.open(mode="r") as file:
+    csvr = csv.reader(file)
+    next(csvr)
+    test_edeps = list()
+    for edep in csvr:
+      test_edeps.append(float(edep[0]))
```
Comment (wkrzemien) [inline t14_extended_source/runAnalysis.py]:
```suggestion
    test_edeps = [float(edep[0]) for edep in csvr]
```

File: t14_extended_source/runAnalysis.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  edep_annihilation, edep_prompt, expected_entries = get_edep_plot_data(tree, source_type, decay_type)
+
+  total_edep_len = len(edep_annihilation) + len(edep_prompt)
+  if expected_entries != total_edep_len:
+    return False
+
+  if decay_type == DecayType.STANDARD:
+    edep_plots[source_strings[source_type]] = edep_annihilation
+    return perform_ks_test(edep_annihilation, GammaType.ANNIHILATIONGAMMA, source_type, decay_type)
+  plot_title = "".join([source_strings[source_type],"Prompt - annihilation gamma"])
```
Comment (wkrzemien) [inline t14_extended_source/runAnalysis.py:181]: Do I understand correctly taht if DecayType is not STANDARD then the plot is created?

File: t14_extended_source/runAnalysis.py [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+  """
+  edep_plots  = dict()
+  check_results = list()
+  check_results.append(analyse_one_file(folder, edep_plots, "pPs.root", SourceType.PARAPOSITRONIUM, DecayType.STANDARD))
+  check_results.append(analyse_one_file(folder, edep_plots, "pPs_prompt.root", SourceType.PARAPOSITRONIUM, DecayType.WITHDEEXCITATION))
+  check_results.append(analyse_one_file(folder, edep_plots, "oPs.root", SourceType.ORTHOPOSITRONIUM, DecayType.STANDARD))
+  check_results.append(analyse_one_file(folder, edep_plots, "oPs_prompt.root", SourceType.ORTHOPOSITRONIUM, DecayType.WITHDEEXCITATION))
+
+  if not all(cr for cr in check_results):
+    return False
```
Comment (wkrzemien) [inline t14_extended_source/runAnalysis.py:237]:
```suggestion
    return  all(cr for cr in check_results):
    
```

**Review by wkrzemien — COMMENTED** [review]
Very nice! I have just a few small comments.
Maybe you could also add a short description what is compared in the KS test. Kind of what is the purpose of the whole test

File: t14_extended_source/runAnalysis.py (hunk 4 above)
Comment (Crauzerus) [reply t14_extended_source/runAnalysis.py:181]: In both cases is created - in first case (STANDARD) as a title is used source_strings[source_type], but for the second type (DecayType.WITHDEEXCITATION) we have to separate edep plots for annihilation and prompt gamma - so we have to set different titles for edep plots.

File: t14_extended_source/runAnalysis.py (hunk 3 above)
Comment (Crauzerus) [reply t14_extended_source/runAnalysis.py]: Done

File: t14_extended_source/runAnalysis.py (hunk 1 above)
Comment (Crauzerus) [reply t14_extended_source/runAnalysis.py]: Done

**Comment by Crauzerus** [comment]
> Is the file t14_extended_source/output.png added as reference?

My mistake - this file should not be here.

File: t14_extended_source/runAnalysis.py (hunk 2 above)
Comment (Crauzerus) [reply t14_extended_source/runAnalysis.py]: Done

**Pull request closed without merging**